src/bika/lims/controlpanel/bika_analysisservices.py

```diff
@@ -53,65 +50,22 @@
 
 class AnalysisServiceCopy(BrowserView):
     template = ViewPageTemplateFile("templates/analysisservice_copy.pt")
-    # should never be copied between services
-    skip_fieldnames = [
-        "UID",
-        "id",
-        "title",
-        "ShortTitle",
-        "Keyword",
-    ]
     created = []
 
-    def create_service(self, src_uid, dst_title, dst_keyword):
-        folder = self.context.bika_setup.bika_analysisservices
-        dst_service = _createObjectByType("AnalysisService", folder, tmpID())
-        # manually set keyword and title
-        dst_service.setKeyword(dst_keyword)
-        dst_service.setTitle(dst_title)
-        dst_service.unmarkCreationFlag()
-        _id = renameAfterCreation(dst_service)
-        dst_service = folder[_id]
-        return dst_service
-
-    def validate_service(self, dst_service):
-        # validate entries
-        validator = ServiceKeywordValidator()
-        # baseschema uses uniquefieldvalidator on title, this is sufficient.
-        res = validator(dst_service.getKeyword(), instance=dst_service)
-        if res is not True:
-            self.savepoint.rollback()
-            self.created = []
+    def copy_service(self, source, title, keyword):
+        """Creates a copy of the given AnalysisService object, but with the
+        given title and keyword
+        """
+        # Validate the keyword
+        err_msg = serviceapi.check_keyword(keyword)
```

Comment:
Better do a try/except block here and raise an error in `serviceapi.check_keyword`